docs/macros/define-stylex.md

```diff
@@ -0,0 +1,131 @@
+# defineStyleX <PackageVersion name="@vue-macros/define-stylex" />
+
+<StabilityLevel level="experimental" />
+
+Define and consume [StyleX](https://stylexjs.com/) styles in `<script setup>`.
+
+|   Features   |     Supported      |
+| :----------: | :----------------: |
+|    Vue 3     | :white_check_mark: |
+|    Nuxt 3    | :white_check_mark: |
+| Volar Plugin | :white_check_mark: |
+|    Vue 2     | :white_check_mark: |
+
+## Setup
+
+To use StyleX, you should install and configure StyleX first.
+
+### Vite
+
+```sh
+pnpm add @stylexjs/stylex vite-plugin-stylex
+```
+
+```ts {4,13}
+import Vue from '@vitejs/plugin-vue'
+import VueMacros from 'unplugin-vue-macros/vite'
+import { defineConfig } from 'vite'
+import styleX from 'vite-plugin-stylex'
+
+export default defineConfig({
+  plugins: [
+    VueMacros({
+      plugins: {
+        vue: Vue(),
+      },
+    }),
+    styleX(), // Must be placed after VueMacros
+  ],
+})
+```
+
+```vue twoslash {9-10}
+<!--App.vue-->
+<script setup lang="ts"></script>
+
+<template>
+  <div>some content</div>
+</template>
+
+<style>
+/* import StyleX stylesheet, according to https://github.com/HorusGoul/vite-plugin-stylex */
+@stylex stylesheet;
+</style>
+```
+
+## Basic Usage
+
+```vue twoslash
+<script setup lang="ts">
+const styles = defineStyleX({
+  red: { color: 'red' },
+})
+</script>
+
+<template>
+  <p v-stylex="styles.red">Red Text</p>
+</template>
+```
+
+which will be transformed to:
```

Comment:
Let's use `<details>` for compiled code, for example: https://vue-macros.dev/macros/define-options.html#basic-usage